### PR TITLE
fix(config): support pass false to plugins

### PR DESCRIPTION
**Pull request by puppet-666**
## Summary

To support pass false to plugins

link #4939 
link #4926 

## Test Plan
This is my first time contributing to rspack. At present, I only know how to create tests in the 'packages \ rspack \ tests \ configCases' directory, but I am not very familiar with the running logic of test cases. Can you give me some help?

## Require Documentation?

- [ x ] No
- [ ] Yes, the corresponding rspack-website PR is \_\_

### CODE REVIEW

**Review by ScriptedAlchemy — DISMISSED** [review]

**Comment by hardfist** [comment]
@puppet-666 please sign the cla

**Comment by puppet-666** [comment]
> @puppet-666 please sign the cla

done

**Comment by hardfist** [comment]
https://web-infra-dev.github.io/rspack-dev-guide/testing/intro.html @puppet-666  you can read this about how rspack test

**Comment by puppet-666** [comment]
> [web-infra-dev.github.io/rspack-dev-guide/testing/intro.html](https://web-infra-dev.github.io/rspack-dev-guide/testing/intro.html) @puppet-666 you can read this about how rspack test

Thanks for your answer. Are test cases required for this merge? If so I will add test cases

**Comment by hardfist** [comment]
@puppet-666  I think we should align this with webpack and allow all falsy value here, you can take https://github.com/webpack/webpack/pull/17339 for reference(no need to support falsy value for loader in this PR)

File: packages/rspack/src/rspack.ts [hunk 1]
```diff
@@ -70,6 +70,8 @@ function createCompiler(userOptions: RspackOptions): Compiler {
 		for (const plugin of options.plugins) {
 			if (typeof plugin === "function") {
 				(plugin as RspackPluginFunction).call(compiler, compiler);
+			} else if (typeof plugin === "boolean") {
```
Comment (hardfist) [inline packages/rspack/src/rspack.ts:73]: also need update the schema check to allow false

File: packages/rspack/src/config/zod.ts [hunk 2]
```diff
@@ -26,6 +26,18 @@ const mode = z.enum(["development", "production", "none"]);
 export type Mode = z.infer<typeof mode>;
 //#endregion
 
+//#region Falsy
+const falsy = z.union([
+	z.literal(false),
+	z.number().int().min(0),
```
Comment (LingyuCoder) [inline packages/rspack/src/config/zod.ts:32]: Perhaps `z.literal(0)` is better?

**Review by hardfist — APPROVED** [review]

**Comment by hardfist** [comment]
thanks!

**Review by LingyuCoder — APPROVED** [review]

**Pull request merged**